### ISSUE TITLE
App router with Next 13.5.1 - Build error on preview routes

### ISSUE THREAD

**Issue by roux1max** [prismicio/prismic-next#84]
Hi!

Thanks for your amazing work on this module. I have a little build problem since I updated to Next 13.5.1.

### Versions

- `@prismicio/next`: 1.3.4
- `next`: 13.5.1
- node: 18.15

### Reproduction
Following Prismic documentation for preview, add a `/preview` route handler in Next.Js App Router with the following content:

```typescript
export async function GET(request: NextRequest) {
    const client = createClient();

    draftMode().enable();

    await redirectToPreviewURL({ client, request });
}
```

### Steps to reproduce

Run a simple build with Next: `npm run build`

### What is expected?

No build error.

### What is actually happening?

Error on checking validity of types:
```shell
src/app/api/preview/route.ts
Type error: Route "src/app/api/preview/route.ts" has an invalid export:
  "Promise<void>" is not a valid GET return type:
    Expected "Response | Promise<Response>", got "Promise<void>".
      Expected "Promise<Response>", got "Promise<void>".
        Expected "Response", got "void".
``` 

The error appeared since Next.JS changed their expected return type for the route handlers. The `redirectToPreviewURL()` should have a return type of `never` to match the call to `redirect()` from `next/navigation`. The problem is also present for `exitPreview()` which should return `Response` instead of `Response | void`.

I get that those problems are due to `@prismicio/next` trying to support both _Pages router_ and _App router_ with the same functions. Maybe it could be interesting to have a parameter on those functions that provides a way to change the function return type conditionnally.

**Comment by angeloashmore** [comment]
Hi @roux1max, thanks for the detailed report! 🙂

The type error should be fixed as of `@prismicio/next` `v1.3.5`. You can find details on the necessary changes in this PR: #85. Your diagnosis was correct; `redirectToPreviewURL()` needed to return `never` when we detect the App Router is being used.

Could you update to the latest version and let me know if the build error still happens?

```sh
npm install @prismicio/next@latest
```

Thank you!

**Comment by roux1max** [comment]
The problem is now fixed!
Thanks for the quick answer!
